output.tf

```diff
@@ -0,0 +1,4 @@
+output "demo_instance_public_ip" {
```

Comment:
for this problem you can output only instance endpoint but you know how to use output variables which is fine